bftengine/src/bcstatetransfer/SourceSelector.hpp

```diff
@@ -106,6 +121,9 @@ class SourceSelector {
   std::vector<uint16_t> actualSources_;
   bool receivedValidBlockFromSrc_;
 
+  uint16_t nominatedPrimary_ = NO_REPLICA;
+  uint16_t prePrepareMsgCounter_ = 0;
+  static constexpr uint16_t MAX_PREPREPARE_MSGS = 10;
```

Comment:
I think it is better to have this a configuration parameter.
Lets say an SRE finds some issue at costumer related to this value, and you want to increase it. It will be much easier for you to tell him to change the bft_config.yaml.
The work is very minimal to have it as a config parameter.
You can use for now internal default configuration, and not even propagate the parameter to bft_config.yaml (we can change it only once later on).